### ISSUE TITLE
[macOS] Compiler should not be hardcoded, as gcc can also be used (and _should_ be used on some platforms)

### ISSUE THREAD

**Issue by barracuda156** [bellard/quickjs#320]
Current Makefile forces Clang for macOS and FreeBSD, this is undesirable and in certain cases breaking.
It is okay to fall back to Clang by default, but provided CC has not been set in the environment and build args.

**Comment by chqrlie** [comment]
I guess we could check for a `CC` environment variable.  Note however that **gcc** on macOS is by default an alias for **clang**.  The gcc toolchain can still be installed manually I suppose.

**Comment by barracuda156** [comment]
@chqrlie Yes, gcc (real one) can be used on macOS. (On some legacy versions it is the only working compiler.)
